fix(docs): update Auth0 third-party auth guide for custom claims (#44853)
supabase/supabase: apps/docs/content/guides/auth/third-party/auth0.mdx

  'SUPABASE_PUBLISHABLE_KEY',
  {
    accessToken: async () => {
      // Use the ID token which reliably includes custom claims.
      const idToken = (await auth0.getIdTokenClaims())?.__raw
      if (!idToken) throw new Error('Missing ID token')
      return idToken

>     },

Author: Verified: idToken is non-nullable (String) in both auth0_flutter and Auth0.Android. The null guard is unnecessary, adding one would be dead code. The TypeScript example needs it because getIdTokenClaims() returns IdToken | undefined, but the mobile SDKs guarantee a non-null idToken on their Credentials objects.

By default, Auth0 JWTs (both access token and ID token) do not contain a `role` claim in them. If you were to send such a JWT to your Supabase project, the `anon` role would be assigned when executing the Postgres query. Most of your app's logic will be accessible by the `authenticated` role.

A recommended approach to do this is to configure the [`onExecutePostLogin` Auth0 Action](https://auth0.com/docs/secure/tokens/json-web-tokens/create-custom-claims#create-custom-claims) which will add the custom claim.

If you are using **ID tokens** (recommended for the TypeScript SPA example above):

> 

Reviewer: It feels redundant a bit to always use the "recommended" signal of clarification. I would be more direct.

```suggestion
If you are using **ID tokens** from the example above:

```

```javascript
exports.onExecutePostLogin = async (event, api) => {
  api.idToken.setCustomClaim('role', 'authenticated')
}
```

> If you are using **access tokens** instead, you must use a namespaced claim because [Auth0 strips non-namespaced custom claims from access tokens](https://auth0.com/docs/troubleshoot/product-lifecycle/past-migrations/custom-claims-migration):

Reviewer: ```suggestion
For **access tokens**, you must use a namespaced claim because [Auth0 strips non-namespaced custom claims](https://auth0.com/docs/troubleshoot/product-lifecycle/past-migrations/custom-claims-migration):
```

```javascript
exports.onExecutePostLogin = async (event, api) => {
  api.accessToken.setCustomClaim('https://your-app.com/role', 'authenticated')
}
```

<Admonition type="caution">

> Using `api.accessToken.setCustomClaim('role', 'authenticated')` without a namespace may appear to work but Auth0 can silently strip the claim. Use ID tokens or namespaced access token claims to avoid this issue.

Reviewer: ```suggestion
Auth0 can silently strip claims without a namespace when using `api.accessToken.setCustomClaim`. We recommend ID tokens or namespaced access token claims to avoid this issue.
```
